### PR TITLE
Standard deduction tests and fixes

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
- Adds the 9 tests from `test_calcfunctions.py`, using the hierarchical format (there's code duplication in the parameter values - but the fix for this is looking a bit complex, I can have a look at it if it's high priority but for now, going with this)
- Fixes bugs in `standard_deduction` - I also found it useful to decompose the variable a bit, adding the basic and aged/blind deductions as their own variables. I think this is the right level of decomposition to aim for.

### CODE REVIEW

File: openfisca_us/variables/finance/tax/outputs.py [hunk 1]
```diff
@@ -3,44 +3,73 @@
...
         basic_if_not_DSI = where(MIDR, 0, STD.amount.filer[MARS])
         basic_stded = where(
-            taxunit("DSI", period), basic_if_DSI, basic_if_not_DSI
+            tax_unit("DSI", period), basic_if_DSI, basic_if_not_DSI
         )
+        return basic_stded
 
-        # calculate extra standard deduction for aged and blind
+
+class aged_blind_standard_deduction(Variable):
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:27]: Current name could be read as the full stded for them
```suggestion
class aged_blind_extra_standard_deduction(Variable):
```

File: openfisca_us/variables/finance/tax/outputs.py [hunk 2]
```diff
@@ -3,44 +3,73 @@
...
+    def formula(tax_unit, period, parameters):
+        STD = parameters(period).tax.deductions.standard
+        MARS = tax_unit("MARS", period)
+        MARSType = MARS.possible_values
         num_extra_stded = (
-            taxunit("blind_head", period) * 1
-            + taxunit("blind_spouse", period) * 1
+            tax_unit("blind_head", period) * 1
+            + tax_unit("blind_spouse", period) * 1
+            + (tax_unit("age_head", period) >= 65) * 1
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:40]: TODO: Make 65 a policy parameter

File: openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml [hunk 3]
```diff
@@ -1,27 +1,296 @@
...
+        blind_spouse: false
+        c19700: 0
+    people:
+      head:
+        age: 45
+      spouse:
+        age: 44
   output:
     standard_deduction: 12000
+- name: Married, allow charity
```
Comment (MaxGhenis) [inline openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml:35]: Could you add a blank line and/or comments describing the tests? Could use some visual breaking up.

File: openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml [hunk 4]
```diff
@@ -1,27 +1,296 @@
...
+        blind_head: false
+        blind_spouse: false
+        c19700: 0
+    people:
+      head:
+        age: 45
+      spouse:
+        age: 44
   output:
     standard_deduction: 12000
```
Comment (MaxGhenis) [inline openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml:34]: Add the blind/aged extra stded?

File: openfisca_us/variables/entity/person.py [hunk 5]
```diff
@@ -16,6 +16,28 @@ class person_weight(Variable):
...
 
+class is_tax_unit_head(Variable):
+    value_type = bool
+    entity = Person
+    label = "Head of tax unit"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        # Use order of input (first)
+        return person.tax_unit.members_position == 0
```
Comment (MaxGhenis) [inline openfisca_us/variables/entity/person.py:27]: is this based on the raw CPS structure?

File: openfisca_us/variables/finance/tax/outputs.py [hunk 6]
```diff
@@ -3,44 +3,73 @@
...
+    def formula(tax_unit, period, parameters):
         STD = parameters(period).tax.deductions.standard
-        MARS = taxunit("MARS", period)
-        MIDR = taxunit("MIDR", period)
-        MARSType = MARS.possible_values
+        MARS = tax_unit("MARS", period)
+        MIDR = tax_unit("MIDR", period)
         c15100_if_DSI = max_(
-            350 + taxunit("earned", period), STD.amount.dependent
+            350 + tax_unit("earned", period), STD.amount.dependent
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:17]: todo: make 350 a policy parameter

File: openfisca_us/variables/finance/tax/outputs.py [hunk 7]
```diff
@@ -3,44 +3,73 @@
...
+
+
+class standard_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = u"Standard deduction"
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        # calculate basic standard deduction
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:58]: I'm guessing this is copied from taxcalc but could we adopt the PEP8 convention of capitalizing the first word in a comment? https://www.python.org/dev/peps/pep-0008/#comments

**Review by MaxGhenis — COMMENTED** [review]
Nice, I agree on splitting out the blind/aged stded

File: openfisca_us/variables/entity/person.py (hunk 5 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/entity/person.py:27]: Not intentionally, just needed a sensible default value. It'll be overridden by input values.

File: openfisca_us/variables/finance/tax/outputs.py (hunk 7 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py:58]: Yeah that's better, and I've done this myself in other places. I'm guessing we should also add this to Black/another formatter to catch this?

Comment (MaxGhenis) [reply openfisca_us/variables/finance/tax/outputs.py:58]: Black doesn't check comments, and I think autopep8 only breaks comments into multiple lines and adds a space after `#`. Automatically capitalizing could violate PEP8 guidance:
> The first word should be capitalized, **unless it is an identifier that begins with a lower case letter (never alter the case of identifiers!).**

But autopep8's could be a start, should we run `autopep8` before `black`? I filed https://github.com/UBICenter/openfisca-us/issues/43 to discuss.

File: openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml [hunk 8]
```diff
@@ -1,27 +1,306 @@
...
+        blind_head: false
+        blind_spouse: false
+        c19700: 0
+    people:
+      head:
+        age: 44
+  output:
+    standard_deduction: 0
+
+- name: Marrid, Single, dep, under earn
```
Comment (MaxGhenis) [inline openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml:256]:
```suggestion
- name: Married, Single, dep, under earn
```

File: openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml (hunk 4 above)
Comment (nikhilwoodruff) [reply openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml:36]: Added to the tests

File: openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml (hunk 3 above)
Comment (nikhilwoodruff) [reply openfisca_us/tests/baseline/finance/tax/deductions/standard.yaml:40]: Added

File: openfisca_us/variables/finance/tax/outputs.py (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py]: Done here, and with 350

File: openfisca_us/variables/finance/tax/outputs.py (hunk 6 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py]: Done!

**Review by MaxGhenis — APPROVED** [review]
good to merge after resolving conflicts

**Comment by MaxGhenis** [comment]
Actually could I suggest a different organization for the [standard deduction policy parameters](https://github.com/UBICenter/openfisca-us/tree/47c6acc6146f3677422eca12909117ce092b85f9/openfisca_us/parameters/tax/deductions/standard)? It's currently this:
* amount
  * age_threshold
  * aged_or_blind
  * dependent
  * dependent_addition
  * filer
* charity
  * allow_nonitemizers
  * nonitemizers_max

How about moving `charity` up a folder to `tax` (since it's not modifying the standard deduction) and then within `standard` doing this?
* aged_or_blind
  * amount
  * age_threshold
* dependent
  * amount
  * additional_earned_income (formerly dependent_addition)
* amount (formerly filer)

**Comment by nikhilwoodruff** [comment]
@MaxGhenis I think that's a better structure, added

File: openfisca_us/variables/finance/tax/outputs.py [hunk 9]
```diff
@@ -59,7 +64,7 @@ class standard_deduction(Variable):
     def formula(tax_unit, period, parameters):
         # Calculate basic standard deduction
         basic_stded = tax_unit("basic_standard_deduction", period)
-        STD = parameters(period).tax.deductions.standard
+        CHARITY = parameters(period).tax.deductions.itemized.charity
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:67]: I understand capitalizing acronyms but this seems to veer into capitalization that should be reserved for constants

**Review by MaxGhenis — COMMENTED** [review]
feel free to merge after addressing this, either by modifying or justifying

File: openfisca_us/variables/finance/tax/outputs.py (hunk 9 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py]: Makes sense, I changed it to lowercase

**Pull request merged**